test/jdk/jdk/incubator/vector/VectorMathTest.java

```diff
@@ -0,0 +1,245 @@
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8338021
+ * @summary Test unsigned and saturating scalar operators for use with Vector API
+ * @modules jdk.incubator.vector
+ * @run testng VectorMathTest
+ */
+
+import jdk.incubator.vector.VectorMath;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.lang.reflect.Array;
+import java.util.function.BinaryOperator;
+
+public class VectorMathTest {
+    // @formatter:off
+    public static final byte  ZERO_B = (byte)0;
+    public static final short ZERO_S = (short)0;
+    public static final int   ZERO_I = 0;
+    public static final long  ZERO_L = 0L;
+
+    public static final byte  TEN_B = (byte)10;
+    public static final int   TEN_S = (short)10;
+    public static final short TEN_I = 10;
+    public static final long  TEN_L = 10L;
+
+    public static final byte  FIFTY_B = (byte)50;
+    public static final int   FIFTY_S = (short)50;
+    public static final short FIFTY_I = 50;
+    public static final long  FIFTY_L = 50L;
+
+    public static final byte  SIXTY_B = (byte)60;
+    public static final int   SIXTY_S = (short)60;
+    public static final short SIXTY_I = 60;
+    public static final long  SIXTY_L = 60L;
+
+    public static final byte  UMAX_B = (byte)-1;
+    public static final short UMAX_S = (short)-1;
+    public static final int   UMAX_I = -1;
+    public static final long  UMAX_L = -1L;
+
+    public static byte[]  INPUT_SB = {Byte.MIN_VALUE,    (byte)(Byte.MIN_VALUE + TEN_B),   ZERO_B, (byte)(Byte.MAX_VALUE - TEN_B),   Byte.MAX_VALUE};
+    public static short[] INPUT_SS = {Short.MIN_VALUE,   (short)(Short.MIN_VALUE + TEN_S), ZERO_S, (short)(Short.MAX_VALUE - TEN_S), Short.MAX_VALUE};
+    public static int[]   INPUT_SI = {Integer.MIN_VALUE, (Integer.MIN_VALUE + TEN_I),      ZERO_I, Integer.MAX_VALUE - TEN_I,        Integer.MAX_VALUE};
+    public static long[]  INPUT_SL = {Long.MIN_VALUE,    Long.MIN_VALUE + TEN_L,           ZERO_L, Long.MAX_VALUE - TEN_L,           Long.MAX_VALUE};
```

Comment:
Ok, now we have 4 or 5 hand-crafted examples. Is that sufficient? Some random values would be nice, then we know that at least eventually we have full coverage.

Reply:
Hand crafter cases contains delimiting and general cases, in short they sufficiently cover entire value range.

Reply:
@PaulSandoz do you think this is sufficient coverage?

Reply:
Please note this test was added just to cover scalar operation validation in VectorMath, automated tests exercise these APIs in fallback implementation anyways.

Reply:
I think the coverage is sufficient for now and we can expand later. The test is written so that it should be possible to more easily refactor and add further dynamically generated test cases. (Note it is deliberately not a test designed to specifically exercise C2 - if/when we add auto vectorization IR tests would be required).